src/components/UpdateModal.js

```diff
@@ -0,0 +1,100 @@
+// @flow
+
+import React, { Component } from 'react'
+import { View, StyleSheet } from 'react-native-web'
+import Text from './Text'
+import Button from './Form/Button'
+import Modal from './Modal'
+import { BASIC_SPACING } from '../styles'
+
+const shell = window.require('electron').shell
+
+type Props = {
+  onError?: () => void,
+  onDismiss?: () => void,
+}
+
+type State = {
+  open: boolean,
+  update: {
+    title?: string,
+    url?: string,
+    description?: string,
+    button?: string,
+  },
+}
+
+const CURRENT_VERSION = '0.4.2'
+const VERSION_URL = 'https://mainframe.com/onyx-version.json'
+
+export default class UpdateModal extends Component<Props, State> {
+  state: State = {
+    open: false,
+    update: {
+      title: 'New version',
+      description: 'A new version of Onyx was released',
+      button: 'Download',
+    },
+  }
+
+  componentDidMount() {
+    fetch(VERSION_URL)
+      .then(response => {
+        if (response.ok) {
+          return response.json()
+        } else {
+          this.props.onError && this.props.onError()
```

Comment:
The error is emitted here but not thrown, so it will enter the next `then()` with undefined `data`. Better `throw` an error here so it enters the `.catch()` logic.